lightningd/pay.c

```diff
@@ -397,12 +397,27 @@ static void report_routing_failure(struct log *log,
 		  type_to_string(tmpctx, struct short_channel_id,
 			  	 &fail->erring_channel),
 		  tal_hex(tmpctx, fail->channel_update));
+
+	/* Strip the type from the channel_update. Due to the specification
+	 * being underspecified, some implementations skipped the type
+	 * prefix. Since we are in the minority we adapt (See #1730 and
+	 * lightningnetwork/lnd#1599 for details). */
+	if (fail->channel_update &&
+	    fromwire_peektype(fail->channel_update) == WIRE_CHANNEL_UPDATE) {
+		assert(tal_bytelen(fail->channel_update) > 2);
+		channel_update = tal_arr(fail, u8, 0);
+		towire(&channel_update, channel_update+2, tal_bytelen(channel_update) - 2);
+	} else {
+		channel_update = fail->channel_update;
+	}
+
 	gossip_msg = towire_gossip_routing_failure(tmpctx,
 						   &fail->erring_node,
 						   &fail->erring_channel,
 						   (u16) fail->failcode,
 						   fail->channel_update);
```

Comment:
You still actually send 'fail->channel_update' though!

You probably want to add a status_debug() on the receiving end and enhance the test to look for it?

Reply:
Ehm, true :disappointed: